pkg/cluster/cluster.go

```diff
@@ -145,7 +148,11 @@ func (c *Cluster) setup() error {
 func (c *Cluster) create() error {
 	c.status.SetPhase(spec.ClusterPhaseCreating)
 	if err := c.updateStatus(); err != nil {
-		return fmt.Errorf("failed to update cluster phase (%v): %v", spec.ClusterPhaseCreating, err)
+		return fmt.Errorf("cluster create: failed to update cluster phase (%v): %v", spec.ClusterPhaseCreating, err)
+	}
+
+	if err := c.gc.CollectCluster(c.cluster.Name, c.cluster.UID); err != nil {
+		return fmt.Errorf("cluster create: failed to clean up conflict resources: %v", err)
```

Comment:
maybe `failed to clean up resources left by the previous cluster with the same name (%s): %v` is more clear?